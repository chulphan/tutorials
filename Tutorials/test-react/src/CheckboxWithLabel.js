import React, { Component } from 'react';

class CheckboxWithLabel extends Component {
    state = {
        isChecked: false
    }

    onChange = () => {
        const { isChecked } = this.state;
        this.setState({
             isChecked: !isChecked 
        });
    }

    render() {
        return (
            <label>
                <input
                    type="checkbox"
                    checked={this.state.isChecked}
                    onChange={this.onChange}
                />
                {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
            </label>
        );
    }
}

export default CheckboxWithLabel;