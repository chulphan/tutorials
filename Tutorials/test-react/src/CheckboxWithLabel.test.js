import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CheckboxWithLabel from './CheckboxWithLabel';

Enzyme.configure({adapter: new Adapter()});

it('render the component completely', () => {
    const checkbox = shallow(<CheckboxWithLabel />);

    checkbox
});

test('CheckboxWithLabel changes the text after click', () => {
    // Render a checkbox with label in the document
    const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off"/>);

    expect(checkbox.text()).toEqual('Off');

    expect(checkbox).toMatchSnapshot();

    checkbox.find('input').simulate('change');

    expect(checkbox.text()).toEqual('On');

    expect(checkbox).toMatchSnapshot();
});

