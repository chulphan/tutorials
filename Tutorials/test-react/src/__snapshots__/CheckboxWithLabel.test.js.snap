// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;

exports[`CheckboxWithLabel changes the text after click 2`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  On
</label>
`;
